Tests/MongoSwiftTests/TestUtils.swift

```diff
@@ -206,3 +206,21 @@ internal func rearrangeDoc(_ input: Document, toLookLike standard: Document) ->
     }
     return output
 }
+
+/// A Nimble matcher for testing BSONValue equality.
+internal func bsonEqual(_ expectedValue: BSONValue?) -> Predicate<BSONValue> {
+    return Predicate { (actualExpression: Expression<BSONValue>) throws -> PredicateResult in
+        let msg = ExpectationMessage.expectedActualValueTo("equal <\(String(describing: expectedValue)))>")
+        if let actualValue = try actualExpression.evaluate() {
+            return PredicateResult(
+                bool: bsonEquals(actualValue, expectedValue!),
```

Comment:
won't this cause a crash if the expected value we pass in is nil? let's follow the logic from the original `equals` matcher [here](https://github.com/Quick/Nimble/blob/c68f5fd88997ab93a53cab07434349621882c817/Sources/Nimble/Matchers/Equal.swift#L7)

Reply:
Ah, yes, it would because of `!`. I was following the Nimble example in their README, but I now see that they include small note after it about how their example does not handle `nil`!!!

I've adapted the original `equals` matcher and tested to see that we are not crashing on any of the permutations of `nil`:
1. `nil`, `non-nil` 
1. `non-nil`, `nil` 
1. `nil`, `nil`

Reply:
💯